### PR TITLE
FE-298 make browsers support official 

### PR DESCRIPTION
**Pull request by rajumsys**
- link http://browserl.ist/  in app not loading page. let me know objections
- Added browserslist dep as, otherwise, `npx browserslist` doesn't follow the rule in `package.json`
- explicitly disallowed all IE

we've about 83% coverage now

Notes: 
- Adding `not dead` isn't working during build (version issue, i guess). but > 1% and last 2 versions combination would probably give us something similar. 

- Browserlist repo suggested not to disallow unknown browsers but in our case our query disallowed browsers like baido, opera mobile, qq browsers. 
- There are some notes in ticket.

### CODE REVIEW

**Comment by rajumsys** [comment]
Seems build is failing due to query ` not < 1%`. Could be a version issue. Need to check why/when that gets executed during build process.

File: public/index.html [hunk 1]
```diff
@@ -84,7 +84,7 @@
           <path class='logo__flame' d="M106.5 101.3c-13.4 10.1-16 28-16.4 40.1C68.8 116.8 129.3 44.6 60.6 0 103.1 55.1 0 109.3 0 190.4c0 31.8 19.9 59.9 63.5 69.6 42.8-9.1 64-37.8 64-69.6 0-47.4-29.5-63.3-21-89.1zM63.6 234c-23.3 0-42.2-18.9-42.2-42.2 0-23.3 18.9-42.2 42.2-42.2 23.3 0 42.2 18.9 42.2 42.2.1 23.3-18.8 42.2-42.2 42.2z"/>
         </svg>
         <h3>Oh no, SparkPost is having trouble loading.</h3>
-        <p>Please <a id='reloadLink'>try again</a> soon.</p>
+        <p>Please <a id='reloadLink'>try again</a> soon. Try using a <a target="_blank" href="http://browserl.ist/?q=%3E+1%25%2C+not+ie+%3C%3D+11%2C+Firefox+ESR%2C+last+2+opera+versions%2C+last+2+edge+versions%2C+last+2+safari+versions%2C+last+2+chrome+versions%2C+last+2+opera+versions%2C+last+2+firefox+versions%2C+last+2+FirefoxAndroid+versions">supported browser</a>.</p>
```
Comment (jasonrhodes) [inline public/index.html]: Neat idea but I think we'd be better off with something more vague -- maybe something like:
```html
<p>
  We're very sorry for the inconvenience. Please try again soon.
  <br /><small>(Note: Some users visiting from a legacy or uncommon browser may want to consider trying another browser or upgrading to the latest version.)</small>
</p>
```

File: package.json [hunk 2]
```diff
@@ -159,10 +160,15 @@
   },
   "browserslist": {
     "production": [
-      ">1%",
-      "last 4 versions",
+      "> 1%",
+      "not ie <= 11",
```
Comment (jasonrhodes) [inline package.json]: maybe just "not ie" ?

**Comment by jasonrhodes** [comment]
What happened to this AC re: Sentry (I saw your comment in the Jira ticket but still not clear why we can't block the IE11 UA string, at least):
"Probably best to blacklist the specific UA string for IE11 to start"

File: public/index.html (hunk 1 above)
Comment (rajumsys) [reply public/index.html]: Yeah, I later felt my copy `Try using...`, assumes they're using old browser which they may not. 

i can update with your copy. but do you want to say 'some users...'. i think it's fine about to talk this particular case like 'if you're using ....'

File: package.json [hunk 3]
```diff
@@ -159,10 +160,15 @@
...
       "Firefox ESR",
-      "not ie < 11"
+      "last 2 opera versions",
+      "last 2 edge versions",
+      "last 2 safari versions",
+      "last 2 chrome versions",
+      "last 2 firefox versions",
+      "last 2 FirefoxAndroid versions"
```
[the remark this answers is not in the document]
Comment (rajumsys) [reply package.json]: yes, it was bringing some browsers which i have no way to test (it's true i haven't tested all anyway though) like baidu, blackberry, Android Browser4.4.3-4.4.4 etc.

File: package.json (hunk 2 above)
Comment (rajumsys) [reply package.json]: no, it wouldn't work. it requires a version :(

**Comment by rajumsys** [comment]
> "Probably best to blacklist the specific UA string for IE11 to start"

Yeah, the comment is the reason I didn't do it. we can block IE11. But why specifically IE11 only?

File: public/index.html (hunk 1 above)
Comment (jasonrhodes) [reply public/index.html]: I like it being as vague as possible, but feel free to tweak what I suggested :)

File: public/index.html [hunk 4]
```diff
@@ -28,21 +28,65 @@
...
+        var isIE = /*@cc_on!@*/false || !!document.documentMode;
+
+        switch(true){
+          // Opera 8.0+
+          case (!!window.opr && !!opr.addons) || !!window.opera || navigator.userAgent.indexOf(' OPR/') >= 0:
+            return 'Opera';
+
+          // Firefox 1.0+
+          case typeof InstallTrigger !== 'undefined':
+          return 'Firefox';
```
Comment (jasonrhodes) [inline public/index.html]: indent?

File: public/index.html [hunk 5]
```diff
@@ -28,21 +28,65 @@
...
+          case !!window.chrome && !!window.chrome.webstore:
+            return 'Chrome';
+
+          case /constructor/i.test(window.HTMLElement) || (function (p) { return p.toString() === "[object SafariRemoteNotification]"; })(!window['safari'] || safari.pushNotification):
+            return 'Safari';
+
+          case isIE:
+            return 'IE';
+
+          case !isIE && !!window.StyleMedia:
```
Comment (jasonrhodes) [inline public/index.html]: Don't need the `!isIE` here since the case above handles that, right?

File: public/index.html [hunk 6]
```diff
@@ -28,21 +28,65 @@
 
      <!-- Pushes app load failure to Sentry -->
     <script>
+      /**
+       * Returns browser's name and version
+       * @credit https://jsfiddle.net/311aLtkz/
+       * @returns {*}
```
Comment (jasonrhodes) [inline public/index.html]: returns what now? haha

File: public/index.html [hunk 7]
```diff
@@ -28,21 +28,65 @@
...
+       */
+      function getAgent(){
+        var isIE = /*@cc_on!@*/false || !!document.documentMode;
+
+        switch(true){
```
Comment (jasonrhodes) [inline public/index.html]: whoa this is weird ...

File: public/index.html (hunk 5 above)
Comment (rajumsys) [reply public/index.html]: oh yeah. that looks redundant!

File: public/index.html (hunk 7 above)
Comment (jasonrhodes) [reply public/index.html]: Sorry that wasn't a super helpful comment. :) `switch (true)` feels like a bit of a hard-to-understand hack to me but for this inlined JS code, I don't care too much, so let's just go with it.

Comment (rajumsys) [reply public/index.html]: yeah, it's a hack. alternate is if...else if.....several.

**Pull request merged**